src/bin/cql-stress-cassandra-stress/java_generate/distribution/sequence.rs

```diff
@@ -55,6 +56,58 @@ impl Distribution for SeqDistribution {
     }
 }
 
+pub struct SeqDistributionFactory {
+    min: i64,
+    max: i64,
+}
+
+impl SeqDistributionFactory {
+    pub fn new(min: i64, max: i64) -> Result<Self> {
+        anyhow::ensure!(
+            min < max,
+            "Upper bound ({}) for sequence distribution is smaller or equal than the lower bound ({}).",
+            max,
+            min
+        );
+
+        Ok(Self { min, max })
+    }
+}
+
+impl DistributionFactory for SeqDistributionFactory {
+    fn create(&self) -> Box<dyn Distribution> {
+        Box::new(SeqDistribution::new(self.min, self.max).unwrap())
+    }
+}
+
+impl SeqDistributionFactory {
+    pub fn parse_from_description(desc: Description<'_>) -> Result<Box<dyn DistributionFactory>> {
+        let build_err_msg = || {
+            format!(
+                "Invalid parameter list for sequence distribution: {:?}",
+                desc.args
+            )
+        };
```

Comment:
This pattern looks a bit tedious - you have to add `with_context` before all question marks - and it's easy to forget about it and introduce a question mark without adding context.

I suggest introducing an inner function like suggested here: https://users.rust-lang.org/t/attach-context-to-all-propagated-errors-with-anyhow/45020/2 so that you only need to take care of attaching context in a single place.

Reply:
This applies to other commits that use this pattern as well.